node/src/components/consensus/era_supervisor.rs

```diff
@@ -696,6 +697,28 @@ where
                 .collect(),
         }
     }
+
+    /// Emits a fatal error if the consensus state is still empty.
+    pub(super) fn shutdown_if_necessary(&self) -> Effects<Event<I>> {
+        let should_emit_error = self
+            .era_supervisor
+            .active_eras
+            .get(&self.era_supervisor.current_era)
+            .and_then(|era| {
+                let any_ref = era.consensus.as_any();
+                any_ref.downcast_ref::<HighwayProtocol<I, ClContext>>()
+            })
+            .map(|protocol| protocol.is_state_empty())
+            .unwrap_or(true);
+        if should_emit_error {
+            fatal!(
+                self.effect_builder,
+                "Consensus shutting down due to inability to participate in the network"
+            )
+        } else {
+            Default::default()
+        }
```

Comment:
Does it have to return anything? Can it return `Unit` and do nothing for the `else` clause?

Reply:
Yes, because we're returning an effect. Theoretically we could just panic directly, but @marc-casperlabs suggested using `EffectBuilder::fatal` instead.

Reply:
But you're using `fatal!` macro and not `effect_builder::fatal`, right?

Reply:
Yes, and it calls `EffectBuilder::fatal` under the hood.